samples/cpp/benchmark_app/utils.cpp

```diff
@@ -198,52 +196,45 @@ void update_device_config_for_virtual_device(const std::string& value,
         for (const auto& it : devices_property) {
             const auto& device_name = it.first;
             const auto& device_value = it.second;
-            if (device_config.find(ov::device::properties.name()) == device_config.end() ||
-                (is_load_config && is_dev_set_property[device_name])) {
-                // Create ov::device::properties with ov::num_stream/ov::inference_precision and
-                // 1. Insert this ov::device::properties into device config if this
-                // ov::device::properties isn't existed. Otherwise,
-                // 2. Replace the existed ov::device::properties within device config.
-                is_dev_set_property[device_name] = false;
-                device_config.erase(device_name);
+            if (device_config.find(ov::device::properties.name()) == device_config.end()) {
                 device_config[ov::device::properties.name()] = ov::AnyMap{};
-                auto& secondary_property = device_config.at(ov::device::properties.name()).as<ov::AnyMap>();
-                secondary_property[device_name] = ov::AnyMap{{property.name(), device_value}};
-            } else {
-                auto& secondary_property = device_config.at(ov::device::properties.name()).as<ov::AnyMap>();
-                if (secondary_property.count(device_name)) {
-                    auto& device_property = secondary_property.at(device_name).as<ov::AnyMap>();
-                    device_property.emplace(property(device_value));
-                } else {
-                    secondary_property[device_name] = ov::AnyMap{{property.name(), device_value}};
-                }
+                device_config[ov::device::properties.name()].as<ov::AnyMap>().insert({device_name, ov::AnyMap{}});
+            }
+
+            // because of legacy API 1.0. eg the config from JSON file.
+            if (device_config[ov::device::properties.name()].is<std::string>()) {
+                device_config[ov::device::properties.name()] =
+                    device_config[ov::device::properties.name()].as<ov::AnyMap>();
             }
+
+            auto& device_properties = device_config[ov::device::properties.name()].as<ov::AnyMap>();
+            if (device_properties.find(device_name) == device_properties.end()) {
+                device_properties.insert({device_name, ov::AnyMap{}});
+            }
+
+            // because of legacy API 1.0. eg the config from JSON file.
+            if (device_properties[device_name].is<std::string>()) {
+                device_properties[device_name] = device_properties[device_name].as<ov::AnyMap>();
+            }
+
+            auto& secondary_property = device_properties[device_name].as<ov::AnyMap>();
+            // overwrite if this config existed
+            secondary_property.erase(property.name());
+            secondary_property.insert({property.name(), device_value});
```

Comment:
Add function to avoid code duplication here and inside `setDeviceProperty`

Reply:
Updated.